src/main/java/com/yahoo/bullet/dsl/converter/JSONBulletRecordConverter.java

```diff
@@ -0,0 +1,130 @@
+
+    private static final long serialVersionUID = -9133702879277054842L;
+    private static final Gson GSON = new GsonBuilder().create();
+
+    /**
+     * Constructs a JSONBulletRecordConverter without a schema.
+     *
+     * @throws BulletDSLException if there is an error creating the converter.
+     */
+    public JSONBulletRecordConverter() throws BulletDSLException {
+        super();
+    }
+
+    /**
+     * Constructs a JSONBulletRecordConverter from a given schema.
+     *
+     * @param schema A schema file that specifies the fields to extract and their types.
+     * @throws BulletDSLException if there is an error creating the converter.
+     */
+    public JSONBulletRecordConverter(String schema) throws BulletDSLException {
+        super(schema);
+    }
+
+    /**
+     * Constructs a JSONBulletRecordConverter from a given configuration.
+     *
+     * @param bulletConfig The configuration that specifies the settings for a JSONBulletRecordConverter.
+     * @throws BulletDSLException if there is an error creating the converter.
+     */
+    public JSONBulletRecordConverter(BulletConfig bulletConfig) throws BulletDSLException  {
+        super(bulletConfig);
+    }
+
+    @Override
+    public BulletRecord convert(Object object, BulletRecord record) throws BulletDSLException {
+        String json = (String) object;
+        Map<String, Object> data = GSON.fromJson(json, new TypeToken<Map<String, Object>>() { }.getType());
+        return super.convert(data, record);
+    }
+
+    @Override
+    protected TypedObject getTypedObject(String name, Type type, Serializable value) {
+        return super.getTypedObject(name, type, fixNumberType(type, value));
+    }
+
+    private static Serializable fixNumberType(Type expected, Serializable object) {
+        // Cannot get nulls as it is already checked before getTypedObject is called in BulletRecordConverter#convert
+        switch (expected) {
+            case INTEGER:
+                return toInt(object);
+            case LONG:
+                return toLong(object);
+            case FLOAT:
+                return toFloat(object);
+            case INTEGER_MAP:
+                return toNumberMap(object, JSONBulletRecordConverter::toInt);
+            case LONG_MAP:
+                return toNumberMap(object, JSONBulletRecordConverter::toLong);
+            case FLOAT_MAP:
+                return toNumberMap(object, JSONBulletRecordConverter::toFloat);
+            case INTEGER_LIST:
+                return toNumberList(object, JSONBulletRecordConverter::toInt);
+            case LONG_LIST:
+                return toNumberList(object, JSONBulletRecordConverter::toLong);
+            case FLOAT_LIST:
+                return toNumberList(object, JSONBulletRecordConverter::toFloat);
+            case INTEGER_MAP_MAP:
+                return toNumberMap(object, s -> toNumberMap(s, JSONBulletRecordConverter::toInt));
```

Comment:
Put null checks in these lambdas (here and in the replaceAll's in toNumberMap and toNumberList) since it's technically possible to have nulls. TypedObject does not check the entire collection when determining the type, I think.